src/treq/response.py

```diff
@@ -95,3 +97,25 @@ def cookies(self):
                 jar.set_cookie(cookie)
 
         return jar
+
+    def fail_for_status(self):
+        """
+        Fail with an error if the response has an error status.
+
+        :rtype: A `Deferred` that fires with the response itself if the status
+            is a known non-error code, or that calls its errback chain with a
+            :class:`twisted.web.error.Error` otherwise.
+        """
+        if _is_error_code(self.code):
+            d = defer.succeed(self)
+        else:
+            d = self.content().addCallback(self._raise_error)
+
+        return d
+
+    def _raise_error(self, body):
+        raise Error(self.code, response=body)
+
+
+def _is_error_code(code):
+    return 100 <= code < 400
```

Comment:
100-series status codes are not included because they are not error codes. I will edit the docstring to clarify that the meaning of 'error status' is derived from the plain meaning of RFC 7231 §6.